dashboard/api_test.py

```diff
@@ -466,23 +505,53 @@ def test_not_paid_in_past_grade_frozen(self, froz_mock, upgr_mock):
             enr_end=self.now-timedelta(weeks=53),
             edx_key="course-v1:MITx+8.MechCX+2014_T1"
         )
-
-        # case when the user has not an entry in the frozen grades
+        # case when the user has not an entry in the frozen grades the code tries to create it
         self.mmtrack.extract_final_grade.side_effect = FinalGrade.DoesNotExist
+        # in case the created frozen grade that says the user passed
+        freeze_mock.return_value = Mock(passed=True)
         run_status = api.get_status_for_courserun(crun, self.mmtrack)
-        assert run_status.status == api.CourseRunStatus.CURRENTLY_ENROLLED
-        # just resetting the mock does not remove the side effect
-        self.mmtrack.extract_final_grade.side_effect = None
+        assert run_status.status == api.CourseRunStatus.CAN_UPGRADE
+        # in case the created frozen grade that says the user did not pass
```

Comment:
Each of these "in case" comments sounds like they're testing something different. If that's true, then they should each be a separate test function.